### PR TITLE
fix(encoder): correct return value on `.write`

### LINKED ISSUE #27: fail to encode array of nested classes
I have 2 classes with custom encoding:

```
class Serializable {
  serialize () {
    const encoder = new cbor.Encoder()
    this.encodeCBOR(encoder)
    return encoder.finalize()
  }

  static deserialize (serialized) {
    return cbor.decodeAll(serialized)
  }
}

class TestA extends Serializable {
  encodeCBOR (gen) {
    return gen.write(44)
  }
}

class TestB extends Serializable {
  encodeCBOR (gen) {
    return gen.write([
      new TestA()
    ])
  }
}
```

This works fine and `TestB` serializes to `<Buffer 81 18 2c>`. If I update `TestB` to this:

```
class TestB extends Serializable {
  encodeCBOR (gen) {
    return gen.write([
      new TestA(),
      true,
    ])
  }
}
```

Then it serializes to `<Buffer 82 18 2c>`, which has the correct length but not the new element. Attempting to decode that throws a `Error: Undeterminated nesting`.

### PR DESCRIPTION
**Pull request by dignifiedquire**
Due to not returning the success value on `.write`, the encoding was
not finished.

Closes #27

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,17 +12,17 @@
...
-    "test:node": "aegir-test --env node",
-    "lint": "aegir-lint",
-    "docs": "aegir-docs",
-    "release": "aegir-release --docs",
-    "release-minor": "aegir-release --type minor --docs",
-    "release-major": "aegir-release --type major --docs",
-    "build": "aegir-build",
-    "coverage": "aegir-coverage",
-    "coverage-publish": "aegir-coverage publish",
+    "test": "aegir test --files test/**/*.spec.js",
```
Comment (daviddias) [inline package.json:15]: is the `--files` flag really needed?

File: package.json [hunk 2]
```diff
@@ -12,17 +12,17 @@
...
+    "test:browser": "aegir test --target browser",
+    "test:node": "aegir test --target node",
+    "lint": "aegir lint",
+    "docs": "aegir docs",
+    "release": "aegir release --docs",
+    "release-minor": "aegir release --type minor",
+    "release-major": "aegir release --type major",
+    "build": "aegir build",
+    "coverage": "aegir coverage",
+    "coverage-publish": "aegir coverage --upload",
```
Comment (daviddias) [inline package.json:25]: use `--provider coveralls`

**Review by daviddias — APPROVED** [review]
Couple of nitpicks, overall LGTM, CI is not happy though. Can we have this module in the ipfs org?

**Comment by FloorLamp** [comment]
Any chance we can get this merged?

**Comment by wanderer** [comment]
bump

**Comment by dignifiedquire** [comment]
fixed and released, sorry for the delay

**Pull request merged**